### PR TITLE
Add support to build ARM64 images

### LINKED ISSUE #225: Require images for ARM64
I am trying to use swift image over ARM64v8 but it seems it does not have an arm64 supported tag in the [docker hub](https://hub.docker.com/_/swift)

I have successfully built the [image ]( https://hub.docker.com/layers/odidev/swift/ubuntu1804/images/sha256-f6f4fabb809b45cb94e7f5d192f5577a9d8bc34a6452584be09d9f81df979b9a?context=repo
)on arm64 platform by using [swift-arm](https://github.com/futurejones/swift-arm64/releases/v5.3.3-RELEASE/) binary for ARM64 platform and implementing these [changes](https://github.com/odidev/swift-docker/commit/3dc5732997c92ad3c9f082b36bf8ace512805946).
ARM64 Image: https://hub.docker.com/layers/odidev/swift/ubuntu1804/images/sha256-f6f4fabb809b45cb94e7f5d192f5577a9d8bc34a6452584be09d9f81df979b9a?context=repo

Changes : https://github.com/odidev/swift-docker/commit/3dc5732997c92ad3c9f082b36bf8ace512805946

Do you have any plans for releasing an ARM64 image?
May I know how the amd64 image is getting published on Docker hub?
If interested, I will raise a PR.

### PR DESCRIPTION
**Pull request by odidev**
Fixes #225 

Modified **5.3/ubuntu/18.04/Dockerfile** to build ARM64 docker images.

Signed-off-by: odidev <odidev@puresoftware.com>

### CODE REVIEW

File: 5.3/ubuntu/18.04/Dockerfile [hunk 1]
```diff
@@ -43,17 +43,23 @@ RUN set -e; \
     && SWIFT_SIG_URL="$SWIFT_BIN_URL.sig" \
     # - Grab curl here so we cache better up above
     && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -q update && apt-get -q install -y curl && rm -rf /var/lib/apt/lists/* \
+    && apt-get -q update && apt-get -q install -y curl sudo wget && rm -rf /var/lib/apt/lists/* \
```
Comment (mbarnach) [inline 5.3/ubuntu/18.04/Dockerfile]: This is not needed, especially `sudo`, right?

File: 5.3/ubuntu/18.04/Dockerfile [hunk 2]
```diff
@@ -43,17 +43,23 @@ RUN set -e; \
...
-    && curl -fsSL "$SWIFT_BIN_URL" -o swift.tar.gz "$SWIFT_SIG_URL" -o swift.tar.gz.sig \
-    && gpg --batch --quiet --keyserver ha.pool.sks-keyservers.net --recv-keys "$SWIFT_SIGNING_KEY" \
-    && gpg --batch --verify swift.tar.gz.sig swift.tar.gz \
-    # - Unpack the toolchain, set libs permissions, and clean up.
-    && tar -xzf swift.tar.gz --directory / --strip-components=1 \
-    && chmod -R o+r /usr/lib/swift \
-    && rm -rf "$GNUPGHOME" swift.tar.gz.sig swift.tar.gz \
-    && apt-get purge --auto-remove -y curl
+    && if [ `uname -m` = "aarch64" ] ; then \
+        wget https://github.com/futurejones/swift-arm64/releases/download/v5.3.3-RELEASE/swiftlang-5.3.3-ubuntu-18.04-release-aarch64-4-2021-02-01.tar.gz && \
```
Comment (mbarnach) [inline 5.3/ubuntu/18.04/Dockerfile]: If the redirection through `curl` is the issue, then 
```
curl -L https://github.com/futurejones/swift-arm64/releases/download/v5.3.3-RELEASE/swiftlang-5.3.3-ubuntu-18.04-release-aarch64-4-2021-02-01.tar.gz --output swiftlang-5.3.3-ubuntu-18.04-release-aarch64-4-2021-02-01.tar.gz && \
```
this is doing the trick for me.

**Comment by mbarnach** [comment]
Nice idea! Unfortunately, I don't think this can be a permanent solution: too manual, and has to be done per supported architectures.
On the Swift Forums, there is a nice discussion about Swift supporting ARM M1 (https://forums.swift.org/t/arm64-swift-docker-images/43864/4). I think we need to raise the attention to that to get it through!

Nevertheless, your PR is a nice quick hack for all of us having an M1 and willing to test it! 🎉

**Comment by odidev** [comment]
Hi @mbarnach
Thanks for the suggestion. I have removed the sudo and wget from installation and used curl to download SWIFT-ARM binaries and have amended the changes.

**Comment by odidev** [comment]
@mbarnach
Can you please review this PR and let me know if any changes are required?

**Comment by mbarnach** [comment]
@odidev It is working fine. No more extra packages.

Nonetheless, this is only for Ubuntu 18.04. As said previously, it is great for testing, but I hope the Swift Core team will add the support for ARM64 soon in their official Docker images.

**Comment by shahmishal** [comment]
Official Docker (hub.docker.com) has stop supporting Ubuntu 18.04, I would recommend using Ubuntu 20.04 or 22.04 if possible.

**Pull request closed without merging**